modules/calib/src/multiview_calibration.cpp

```diff
@@ -19,11 +20,11 @@ class RobustExpFunction : public RobustFunction {
     explicit RobustExpFunction (float scale_=30.0f) : over_scale(-1.442695040f /scale_) {}
     // err > 0
     float getError(float err) const override {
-        const auto under_exp = err * over_scale;
+        const float under_exp = err * over_scale;
         if (under_exp < -20) return 0; // prevent overflow further
         // http://www.machinedlearnings.com/2011/06/fast-approximate-logarithm-exponential.html
-        union { uint32_t i; float f; } exp_val = { static_cast<uint32_t>(pow_23 * (under_exp + 126.94269504f)) };
-        return err * exp_val.f;
+        softfloat vexp = softfloat::fromRaw(static_cast<uint32_t>(pow_23 * (under_exp + 126.94269504f)));
+        return err * float(vexp);
```

Comment:
float cast does the same thing if you look at the implementation :(
you can do something like:

```cpp
float f = err;
int32_t i = 0;
std::memcpy(&i, &f, sizeof i);
```

It should generate the same assembly.
Now that I think about it, `softfloat` should be fixed as well :)

Reply:
There is `Cv32suf` for that purpose.

Reply:
Exactly,  there is the same bug with strict aliasing

Reply:
OK, I'm not sure if we need this optimization at all.
If there were some performance measurements that justify the usage of that function, then I'd like to see them.
If not, I'll leave it turned off with `#define` and use good, non-optimized version instead.